test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,597 @@
+        });
+    });
+
+
+    describe("Single mod without dependencies", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-mod-2.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-mod-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const combo = addMockPackage({
+                name: "mod",
+                versions: [
+                    {version: "2.0.0"},
+                    {version: "1.0.0"}
+                ]
+            });
+
+            const combos = [combo];
+
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("Single mod with a dependency", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modWithADependency-2.0.0", "author-mod-2.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modWithADependency-2.0.0", "author-mod-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const mod = addMockPackage({
+                name: "mod",
+                versions: [
+                    {version: "2.0.0"},
+                    {version: "1.0.0"}
+                ]
+            });
+            const comboWithADependency = addMockPackage({
+                name: "modWithADependency",
+                dependencies: ["author-mod-1.0.0"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-mod-2.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-mod-1.0.0"]}
+                ]
+            });
+            const combos = [comboWithADependency];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod pack with a dependency", () => {
+        it.each([
+            [InstallMode.INSTALL_SPECIFIC, ["author-modPack-2.0.0", "author-mod-1.0.0"]],
+            [InstallMode.UPDATE_ALL, ["author-modPack-2.0.0", "author-mod-2.0.0"]],
+        ])
+        ("", async (installMode, expected) => {
+            const mod = addMockPackage({
+                name: "mod",
+                versions: [
+                    {version: "2.0.0"},
+                    {version: "1.0.0"}
+                ]
+            });
+            const modPack = addMockPackage({
+                name: "modPack",
+                dependencies: ["author-mod-1.0.0"],
+                categories: ["Modpacks"],
+                versions: [
+                    {version: "2.0.0", dependencies: ["author-mod-1.0.0"]},
+                    {version: "1.0.0", dependencies: ["author-mod-1.0.0"]}
+                ]
+            });
+
+            const combos = [modPack];
+            expectToContainOnly(await getFullDependencyList(combos, game, [], installMode), expected);
+        });
+    });
+
+
+    describe("A mod pack with a dependency, with a newer version of the dependency already installed", () => {
```

Comment:
For clarity, this test case could be moved under the test case on L560, so the similar tests are grouped together.